ee/desktop/runner/runner_linux.go

```diff
@@ -54,5 +59,73 @@ func runAsUser(uid string, cmd *exec.Cmd) error {
 		},
 	}
 
+	// Set any necessary environment variables on the command (like DISPLAY)
+	envVars := r.userEnvVars(uid)
+	for k, v := range envVars {
+		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", k, v))
+	}
+
 	return cmd.Start()
 }
+
+func (r *DesktopUsersProcessesRunner) userEnvVars(uid string) map[string]string {
+	envVars := make(map[string]string)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	// Get the user's session so we can get their display.
+	sessionOutput, err := exec.CommandContext(ctx, "loginctl", "show-user", uid, "--value", "--property=Sessions").Output()
+	if err != nil {
+		level.Debug(r.logger).Log(
+			"msg", "could not get user session",
+			"uid", uid,
+			"err", err,
+		)
+		return envVars
+	}
+
+	sessions := strings.Trim(string(sessionOutput), "\n")
+	if sessions != "" {
```

Comment:
early return on `sessions =- ""` will be a little cleaner

Reply:
Oh yup for sure, will update!